tests/framework/db/mysql/connection/DeadLockTest.php

```diff
@@ -119,7 +119,12 @@ public function testDeadlockException()
                 . ($logContent ? ". Shared children log:\n$logContent" : '')
             );
         }
-        $this->assertEquals(1, $deadlockHitCount, "exactly one child must hit deadlock; shared children log:\n" . $logContent);
+
+        if (version_compare($this->getConnection()->getSchema()->getServerVersion(), '8.0', '<')) {
+            $this->assertEquals(1, $deadlockHitCount, "exactly one child must hit deadlock; shared children log:\n" . $logContent);
+        } else {
+            $this->assertEquals(0, $deadlockHitCount, "exactly zero children must hit deadlock; shared children log:\n" . $logContent);
```

Comment:
Are you sure that we should expect no deadlock here? In the test we are specifically trying to get it.

Reply:
After `Mysql 8.0.17` this behavior was changed, and the test will fail, while on Mysql 5.7 it runs correctly, currently the ci skips the test because it runs on PHP 7.4, and in build it runs `Mysql 5.7`.

Reply:
So the behavior was changed in after MySQL 8.0.17 and you are sure that the assert you've added is correct for further versions?